### PR TITLE
Section 4 complete to be Reviewed

### PR DESCRIPTION
**Pull request by Jamtartarenice**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -48,4 +49,34 @@ describe('/api (getting all endpoints)', () => {
...
+    test('making sure that its responding with an article with the desired id', () => {
+        return request(app)
+        .get('/api/articles/2')
+        .expect(200)
+        .then((article) => {
+            expect(article.body.article.article_id).toBe(2);
+        });
+    });
+
+    test('making sure that its responding with an article with the desired id when tested multipul times', () => {
```
Comment (saimakhanom) [inline __tests__/app.test.js]: We dont really need this test, it's just doing the same as the one above

File: __tests__/app.test.js [hunk 2]
```diff
@@ -48,4 +49,34 @@ describe('/api (getting all endpoints)', () => {
...
+
+describe('/api/articles/:article_id', () => {
+    test('making sure that its responding with an article with the desired id', () => {
+        return request(app)
+        .get('/api/articles/2')
+        .expect(200)
+        .then((article) => {
+            expect(article.body.article.article_id).toBe(2);
+        });
+    });
```
Comment (saimakhanom) [inline __tests__/app.test.js:72]: we should also check that the article has all the desired properties that we would expect it to have

File: __tests__/app.test.js [hunk 3]
```diff
@@ -48,4 +49,34 @@ describe('/api (getting all endpoints)', () => {
...
+    test('making sure that its responding with an article with the desired id when tested multipul times', () => {
+        return request(app)
+        .get('/api/articles/5')
+        .expect(200)
+        .then((article) => {
+            expect(article.body.article.article_id).toBe(5);
+        });
+    });
+
+    test('making sure that its when it fails it give a special error', () => {
```
Comment (saimakhanom) [inline __tests__/app.test.js]: what kind of fail is happening here? from reading the test description we should know what kind of request we are expecting to fail. try to also have the expected status code in your test descriptions: e.g. "400 - invalid article_id" and "404: valid but non-existent article_id"

File: __tests__/app.test.js [hunk 4]
```diff
@@ -48,4 +49,36 @@ describe('/api (getting all endpoints)', () => {
...
 });
+
+describe('/api/articles/:article_id', () => {
+    test('making sure that its responding with an article with the desired id', () => {
+        return request(app)
+        .get('/api/articles/2')
+        .expect(200)
+        .then((article) => {
+            expect(article.body.article.article_id).toBe(2);
+            expect(Object.keys(article.body.article))
```
Comment (MRombi) [inline __tests__/app.test.js]: Ideally we would want to know what values are there too, we can either check for specific values or for the data type of the value (I.E. expect any number type of syntax). You could use objectContaining as well 🙂

File: __tests__/app.test.js [hunk 5]
```diff
@@ -57,24 +57,26 @@ describe('/api/articles/:article_id', () => {
...
+            [
+                'article_id',
+                'title',
+                'topic',
+                'author',
+                'body',
+                'created_at',
+                'votes',
+                'article_img_url'
+            ]);
```
Comment (MRombi) [inline __tests__/app.test.js]: We should add a test for when article_id is not a number, with status 400 🙂

**Review by MRombi — CHANGES_REQUESTED** [review]
Well done on improving the code from the previous feedback, we are just missing a bit more. Please, action on the feedback I left and resubmit 🙂

**Pull request merged**